From 39da6232ca947a4000afdfd5e10d2bb756bcda88 Mon Sep 17 00:00:00 2001
Message-Id: <73851f5f37d977081224c5274824f138177a9193.1666072612.git.demi@invisiblethingslab.com>
In-Reply-To: <9d96ef018178fb73bc4a4aba121d49e453cc9796.1666072612.git.demi@invisiblethingslab.com>
References: <9d96ef018178fb73bc4a4aba121d49e453cc9796.1666072612.git.demi@invisiblethingslab.com>
From: Demi Marie Obenour <demi@invisiblethingslab.com>
Date: Fri, 1 Apr 2022 18:16:24 -0400
Subject: [PATCH 4/6] Drop support for very old kernels

All kernels Qubes OS uses are new enough to have the dm/ subdirectory.
The path for ancient kernels (before 2.6.31) did not check for dmsetup
failing.
---
 udev/10-dm.rules.in | 17 ++++-------------
 1 file changed, 4 insertions(+), 13 deletions(-)

diff --git a/udev/10-dm.rules.in b/udev/10-dm.rules.in
index 9fae8df95d1f2f36a4f7a4d92201e88f11aa6120..93b7d10e16387d3c6177ea18e2b14e8f0c4be8e5 100644
--- a/udev/10-dm.rules.in
+++ b/udev/10-dm.rules.in
@@ -47,9 +47,6 @@ KERNEL!="dm-[0-9]*", GOTO="dm_end"
 ACTION!="add|change", GOTO="dm_end"
 
 # Decode udev control flags and set environment variables appropriately.
-# These flags are encoded in DM_COOKIE variable that was introduced in
-# kernel version 2.6.31. Therefore, we can use this feature with
-# kernels >= 2.6.31 only. Cookie is not decoded for remove event.
 ENV{DM_COOKIE}=="?*", IMPORT{program}!="(DM_EXEC)/dmsetup udevflags $env{DM_COOKIE}", GOTO="dm_disable"
 
 # Rule out easy-to-detect inappropriate events first.
@@ -104,16 +101,10 @@ LABEL="dm_no_coldplug"
 #   |_ dev still not active            0                        0
 #   \_ dev already active              1                        0
 
-# "dm" sysfs subdirectory is available in newer versions of DM
-# only (kernels >= 2.6.29). We have to check for its existence
-# and use dmsetup tool instead to get the DM name, uuid and 
-# suspended state if the "dm" subdirectory is not present.
-# The "suspended" item was added even later (kernels >= 2.6.31),
-# so we also have to call dmsetup if the kernel version used
-# is in between these releases.
-TEST=="dm", ENV{DM_NAME}="$attr{dm/name}", ENV{DM_UUID}="$attr{dm/uuid}", ENV{DM_SUSPENDED}="$attr{dm/suspended}"
-TEST!="dm", IMPORT{program}="(DM_EXEC)/dmsetup info -j %M -m %m -c --nameprefixes --noheadings --rows -o name,uuid,suspended"
-ENV{DM_SUSPENDED}!="?*", IMPORT{program}="(DM_EXEC)/dmsetup info -j %M -m %m -c --nameprefixes --noheadings --rows -o suspended"
+# Kernels older than 2.6.31 do not have dm/suspended sysfs attribute and are not
+# supported
+ENV{DM_NAME}="$attr{dm/name}", ENV{DM_UUID}="$attr{dm/uuid}", ENV{DM_SUSPENDED}="$attr{dm/suspended}"
+ENV{DM_SUSPENDED}!="?*", GOTO="dm_disable"
 
 # dmsetup tool provides suspended state information in textual
 # form with values "Suspended"/"Active". We translate it to
-- 
Sincerely,
Demi Marie Obenour (she/her/hers)
Invisible Things Lab

